skyportal/handlers/api/allocation.py

```diff
@@ -0,0 +1,176 @@
+from ..base import BaseHandler
+from ...models import DBSession, Group, Allocation, Instrument
+from baselayer.app.access import auth_or_token, permissions
+from marshmallow.exceptions import ValidationError
+
+
+class AllocationHandler(BaseHandler):
+    @auth_or_token
+    def get(self, allocation_id=None):
+        """
+        ---
+        single:
+          description: Retrieve an allocation
+          parameters:
+            - in: path
+              name: allocation_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+               content:
+                application/json:
+                  schema: SingleAllocation
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all allocations
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfAllocations
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        # get owned allocations
+        allocations = (
+            DBSession()
+            .query(Allocation)
+            .filter(
+                Allocation.group_id.in_(
+                    [g.id for g in self.current_user.accessible_groups]
+                )
+            )
+        )
+
+        if allocation_id is not None:
+            try:
+                allocation_id = int(allocation_id)
+            except ValueError:
+                return self.error("Allocation ID must be an integer.")
+            allocations = allocations.filter(Allocation.id == allocation_id).all()
+            if len(allocations) == 0:
+                return self.error("Could not retrieve allocation.")
+            return self.success(data=allocations[0])
```

Comment:
Why doesn't this need to be passed to `schema.dump` as with the multiple case below? I think it was before. Is that really needed below? `BaseHandler.success` does convert everything to JSON already

Reply:
fixed